src/components/NewsList.jsx

```diff
@@ -0,0 +1,42 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+const NewsContent = (props) => {
+  const numEntries = props.numEntries
+    ? props.numEntries
+    : props.articles.length;
+  const colorStyle = (index) => ({
+    color: index % 2 == 0 ? "var(--color-accent-eng)" : "var(--color-accent-math)",
+  });
+
+  return props.articles.splice(0, numEntries).map((article, i) => (
```

Comment:
don't modify args

use `slice` instead of splice

Reply:
good catch, fixed